source/MaterialXRuntime/Private/Commands/PvtRemoveMetadataCmd.cpp

```diff
@@ -0,0 +1,111 @@
+//
+// TM & (c) 2020 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#include <MaterialXRuntime/Private/Commands/PvtRemoveMetadataCmd.h>
+#include <MaterialXRuntime/Private/PvtCommand.h>
+
+#include <MaterialXRuntime/RtPrim.h>
+#include <MaterialXRuntime/RtTypeDef.h>
+#include <MaterialXRuntime/RtValue.h>
+
+namespace MaterialX
+{
+
+PvtRemoveMetadataCmd::PvtRemoveMetadataCmd(const RtObject& obj, const RtToken& name)
+    : PvtCommand()
+    , _obj(obj)
+    , _name(name)
+    , _oldValue()
+{
+}
+
+PvtCommandPtr PvtRemoveMetadataCmd::create(const RtObject& obj, const RtToken& name)
+{
+    return std::make_shared<PvtRemoveMetadataCmd>(obj, name);
+}
+
+void PvtRemoveMetadataCmd::execute(RtCommandResult& result)
+{
+    if (_obj.isValid())
+    {
+        try
+        {
+            if (_obj.getMetadata(_name, RtType::STRING))
+            {
+                // Send message that the metadata is being removed
+                msg().sendRemoveMetadataMessage(_obj, _name);
+
+                // Save old value for undo/redo
+                _oldValue = RtValue::clone(RtType::STRING, _obj.getMetadata(_name, RtType::STRING)->getValue(), _obj.getParent());
```

Comment:
Here you can avoid the second search (getMetadata call) by saving the result from the first call above.